File: Haskell/src/Util/Lista.hs
module Util.Lista (atualizaIndice) where

atualizaIndice :: Int -> a -> [a] -> [a]
atualizaIndice 0 novoValor (_:xs) = novoValor : xs
atualizaIndice n novoValor (x:xs) = x : atualizaIndice (n-1) novoValor xs
atualizaIndice _ _   []      = []

File: Prolog/src/utils/lista.pl

:- module(lista, [
    atualiza_indice/4
]).

atualiza_indice(Indice, NovoElemento, ListaIn, ListaOut) :-
    nth0(Indice, ListaIn, _, Restante),
    nth0(Indice, ListaOut, NovoElemento, Restante).
